src/serializer/ResidualHeapSerializer.js

```diff
@@ -1117,9 +1126,106 @@ export class ResidualHeapSerializer {
         undelay();
       });
     }
+    let classProperties = instance.classProperties;
+
+    if (classProperties !== undefined) {
+      // handle the constructor for the class
+      if (classProperties.methodType === "constructor") {
+        invariant(val.$HomeObject instanceof ObjectValue);
+        let properties = new Map(val.properties);
+
+        // we visit all of these as these, as they form part of the constructor for the class
+        // but we want to serialize them out as class methods in the ResidualFunctions
+        // rather than via the typical Prepack route of serializing properties
+        for (let [key] of val.properties) {
+          if (!this.residualHeapInspector.canIgnoreProperty(val, key)) {
+            let propertyValue = Get(this.realm, val, key);
+
+            if (key === "arguments" || key === "length" || key === "name" || key === "caller" || key === "prototype") {
+              properties.delete(key);
+              this.serializedValues.add(propertyValue);
+            }
+          }
+        }
 
-    undelay();
+        let handleClassMethodValue = (classMethod, methodProps: MethodProperties) => {
+          invariant(classMethod instanceof FunctionValue);
+          // skip processing if its the constructor
+          if (classMethod !== val) {
+            this.serializedValues.add(classMethod);
+            this._serializeValueFunction(classMethod, methodProps);
```

Comment:
@NTillmann, do you think there's a better way to do this/can you think of any problems with not going through serializeValue?

Reply:
Can there be any other reference to the `classMethod`? If so, then that could cause problems.

Reply:
We should try to find a test case for that.

Reply:
In serializeValue, you could log a nice user-facing error if such a `classMethod` ever gets there.